_studio/shared/umc/codec/h265_dec/src/umc_h265_mfx_utils.cpp

```diff
@@ -922,11 +936,21 @@ mfxStatus Query_H265(VideoCORE *core, mfxVideoParam *in, mfxVideoParam *out, eMF
 }
 
 // Validate input parameters
-bool CheckVideoParam_H265(mfxVideoParam *in, eMFXHWType /* type */)
+bool CheckVideoParam_H265(mfxVideoParam *in, eMFXHWType type)
 {
     if (!in)
         return false;
 
+#if defined (MFX_ENABLE_CPLIB)
+    if (in->Protected)
+    {
+        if (type == MFX_HW_UNKNOWN || !IS_PROTECTION_CENC(in->Protected))
+            return false;
+    }
+#else
+    (void)type;
```

Comment:
no need in #else. use unconditionally in the beginning of the function

Reply:
fixed